tests/vector/test_async_vector_env.py

```diff
@@ -28,11 +28,11 @@ def test_create_async_vector_env(shared_memory):
 @pytest.mark.parametrize("shared_memory", [True, False])
 def test_reset_async_vector_env(shared_memory):
     env_fns = [make_env("CartPole-v1", i) for i in range(8)]
-    try:
-        env = AsyncVectorEnv(env_fns, shared_memory=shared_memory)
-        observations = env.reset()
-    finally:
-        env.close()
+
+    env = AsyncVectorEnv(env_fns, shared_memory=shared_memory)
```

Comment:
This is one of the situations where it might be better to do a pyright ignore, or maybe something more clever.

Basically what might (maybe?) happen here is that something crashes inside the environment, the test gets aborted, and then the environment is never properly closed, which can lead to a memory leak, because there are some hanging processes. The `finally` clause prevents that.

It might be possible by using a feature that's existed in gym very quietly, which is using an env as a context manager:
```python
with AsyncVectorEnv(env_fns, shared_memory=shared_memory) as env:
    ... # do whatever
```
I'm not 100% sure, but I think this will make sure to close the environment even if it crashes

Reply:
(on second thought, I'm not sure if this is implemented for VectorEnv, but it definitely is for normal envs)